data-prepper-plugins/opensearch-source/src/test/java/org/opensearch/dataprepper/plugins/source/opensearch/OpenSearchClientRefresherTest.java

```diff
@@ -0,0 +1,81 @@
+package org.opensearch.dataprepper.plugins.source.opensearch;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.opensearch.client.opensearch.OpenSearchClient;
+import org.opensearch.dataprepper.plugins.source.opensearch.worker.client.OpenSearchClientFactory;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class OpenSearchClientRefresherTest {
+    private static final String TEST_USERNAME = "test_user";
+    private static final String TEST_PASSWORD = "test_password";
+
+    @Mock
+    private OpenSearchClientFactory openSearchClientFactory;
+
+    @Mock
+    private OpenSearchSourceConfiguration openSearchSourceConfiguration;
+
+    @Mock
+    private OpenSearchClient openSearchClient;
+
+    private OpenSearchClientRefresher objectUnderTest;
+
+    @BeforeEach
+    void setup() {
+        when(openSearchClientFactory.provideOpenSearchClient(eq(openSearchSourceConfiguration)))
+                .thenReturn(openSearchClient);
+        objectUnderTest = new OpenSearchClientRefresher(openSearchClientFactory, openSearchSourceConfiguration);
```

Comment:
We should use `createObjectUnderTest()` methods instead of creating in the setup method. Using a function and creating in the test allows tests to make changes prior to calling the constructor.